source/extensions/transport_sockets/http_11_proxy/connect.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "envoy/network/connection.h"
+#include "envoy/network/transport_socket.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/common/logger.h"
+#include "source/extensions/transport_sockets/common/passthrough.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Http11Connect {
+
+// If the transport socket options contain proxyInfo and the transport is
+// secure, this will prepend a CONNECT request to the outbound data and strip
+// the CONNECT response from the inbound data.
+class UpstreamHttp11ConnectSocket : public TransportSockets::PassthroughSocket,
+                                    public Logger::Loggable<Logger::Id::connection> {
+public:
+  static const std::string& key();
+
+  UpstreamHttp11ConnectSocket(Network::TransportSocketPtr&& transport_socket,
+                              Network::TransportSocketOptionsConstSharedPtr options);
+
+  void setTransportSocketCallbacks(Network::TransportSocketCallbacks& callbacks) override;
+  Network::IoResult doWrite(Buffer::Instance& buffer, bool end_stream) override;
+  Network::IoResult doRead(Buffer::Instance& buffer) override;
+
+private:
+  void generateHeader();
+  Network::IoResult writeHeader();
+
+  Network::TransportSocketOptionsConstSharedPtr options_;
+  Network::TransportSocketCallbacks* callbacks_{};
+  Buffer::OwnedImpl header_buffer_{};
+  bool stripped_connect_{};
```

Comment:
nit: comment? Also, I think = false would be slightly clearer than {}, but up to you. (I'd also recommend = nullptr for callbacks_ and would be inclined to remove {} for header_buffer_)

Reply:
pretty sure I've been corrected to not use {false} so leaving as is